### PR TITLE
HTTP/2: add initial support

### PR DESCRIPTION
**Pull request by Kriechi**
* refactor pathod and pathoc to support HTTP/2
* split `cmdline` for pathoc and pathod into separate files
* prepare `protocol` usage
    * future TODO: refactor websockets and HTTP/1 into a protocol
* HTTP/2 currently supports:
    * pathoc: 
      * `GET:/` - a GET method and path == "/"
      * `GET:/index.html` - method and path
      * `GET:/:bfoobar` - with body == "foobar"
      * `GET:/:h"foo"="bar"` - with header "foo" = "bar" (multiple headers allowed)
    * pathod:
      * `200` - status code 200
      * `200:bfoobar` - status code 200 with body == "foobar"
      * `200:h"foo"="bar"` - with header "foo" = "bar" (multiple headers allowed)

This PR requires https://github.com/mitmproxy/netlib/pull/70.

### CODE REVIEW

File: libpathod/utils.py [hunk 1]
```diff
@@ -3,10 +3,12 @@
 from netlib import tcp
 
 SSLVERSIONS = {
```
Comment (mhils) [inline libpathod/utils.py]: Does it make sense to move mitmproxy's [`version_to_openssl`](https://github.com/mitmproxy/mitmproxy/blob/36d7e3fcd645ec9b5a7466046e044a2f1c1167ad/libmproxy/proxy/config.py#L97-L122) into netlib and re-use this?

File: libpathod/pathod.py [hunk 2]
```diff
@@ -255,7 +265,7 @@ def handle_http_request(self):
                     path=path,
                     method=method,
                     headers=headers.lst,
-                    httpversion=httpversion,
```
Comment (mhils) [inline libpathod/pathod.py]: Intentional? (According to the comment, this method should support HTTP 1 as well)

Comment (Kriechi) [inline libpathod/pathod.py]: no - already fixed with a new commit.

File: libpathod/utils.py (hunk 1 above)
Comment (Kriechi) [inline libpathod/utils.py]: Yes, sounds like a good idea for a future refactoring.
I'll add a `TODO`.

Comment (mhils) [inline libpathod/utils.py]: :+1:

**Pull request merged**